### PR TITLE
[FLINK-15988][json] Make JsonRowSerializationSchema's constructor private

### PR DESCRIPTION
**Pull request by libenchao**
## What is the purpose of the change

Change JsonRowSerializationSchema's constructor private, and force to use Builder to build JsonRowSerializationSchema.

## Brief change log

- JsonRowSerializationSchema's constructor private
- Add static builder method for JsonRowSerializationSchema
- deprecate JsonRowSerializationSchema.Builder's public constructor

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (**yes** / no)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonRowSerializationSchema.java [hunk 1]
```diff
@@ -99,35 +94,66 @@ public JsonRowSerializationSchema(TypeInformation<Row> typeInfo) {
...
 
-		private final RowTypeInfo typeInfo;
+		private RowTypeInfo typeInfo;
+
+		private Builder() {
+			// private constructor
+		}
 
 		/**
+		 * @deprecated Use {@link JsonRowSerializationSchema#builder()} instead.
```
Comment (wuchong) [inline flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonRowSerializationSchema.java]: Put the `@deprecated` comment after `@param`.

File: flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonRowSerializationSchema.java [hunk 2]
```diff
@@ -99,35 +94,66 @@ public JsonRowSerializationSchema(TypeInformation<Row> typeInfo) {
...
+		 * Finalizes the configuration and checks validity.
+		 * @return Configured {@link JsonRowSerializationSchema}
+		 */
 		public JsonRowSerializationSchema build() {
+			checkArgument(typeInfo != null, "typeInfo should be set.");
 			return new JsonRowSerializationSchema(typeInfo);
 		}
 	}
 
+	public static Builder builder() {
```
Comment (wuchong) [inline flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonRowSerializationSchema.java]: Add a javadoc on this method.

**Comment by libenchao** [comment]
@wuchong Thanks for your review, I've addressed your comments.

**Comment by wuchong** [comment]
The compile is failed, please have a look.

**Comment by libenchao** [comment]
@wuchong fixed check style problem. will do the check before pushing code to github next time.

**Pull request merged**